### ISSUE TITLE
BigQuery: Exception on `ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW`

### ISSUE THREAD

**Issue by VictorLeP** [sqlfluff/sqlfluff#1587]
Query: https://github.com/HTTPArchive/almanac.httparchive.org/blob/34cc2f70c9a60e4bbabcc7da8c719db9b0066fa1/sql/2021/privacy/data_breaches_pwned_websites_per_date_and_breach_sensitivity.sql

The `ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW` line triggers an exception:
```
WARNING    Unable to lint sql/2021/privacy/data_breaches_pwned_websites_per_date_and_class.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/linter/runner.py", line 97, in run
    yield partial()
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/linter/linter.py", line 521, in lint_rendered
    parsed = cls.parse_rendered(rendered)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/linter/linter.py", line 311, in parse_rendered
    parsed, pvs = cls._parse_tokens(
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/linter/linter.py", line 188, in _parse_tokens
    parsed: Optional[BaseSegment] = parser.parse(
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/parser.py", line 32, in parse
    parsed = root_segment.parse(parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 870, in parse
    self.segments = self.expand(self.segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 276, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 42, in parse
    new_self = super().parse(parse_context, parse_grammar=self._parse_grammar)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 811, in parse
    m = parse_grammar.match(segments=segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 130, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/base.py", line 854, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/base.py", line 466, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 130, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 198, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/base.py", line 217, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 130, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 198, in match
    match = self._match_once(unmatched_segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 142, in _match_once
    match, _ = self._longest_trimmed_match(
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/base.py", line 217, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 130, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 90, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/base.py", line 826, in match
    elem = self._get_elem(dialect=parse_context.dialect)
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/parser/grammar/base.py", line 805, in _get_elem
    return dialect.ref(self._get_ref())
  File "/home/victor/Projects/web-almanac/venv/lib/python3.9/site-packages/sqlfluff/core/dialects/base.py", line 232, in ref
    raise RuntimeError(
RuntimeError: Grammar refers to 'RowKeywordSegment' which was not found in the bigquery dialect
```

I copied this line straight from [BigQuery documentation](https://cloud.google.com/bigquery/docs/reference/standard-sql/analytic-function-concepts#compute_a_cumulative_sum), and the query runs fine in the SQL Workspace, so I'm assuming it's valid BigQuery SQL.

## Dialect
bigquery

## Version
sqlfluff version 0.6.9, Python 3.9.5

## Configuration
https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/.sqlfluff
